examples/multimodal/wds.py

```diff
@@ -1,42 +1,34 @@
 from datachain import C, DataChain
 from datachain.lib.webdataset import process_webdataset
 from datachain.lib.webdataset_laion import WDSLaion, process_laion_meta
+from datachain.sql.functions import path
 
-wds = (
+wds_images = (
     DataChain.from_storage("gs://datachain-demo/datacomp-small/shards")
-    .filter(C("file.name").glob("00000000.tar"))
+    .filter(C("file.name").glob("000000[0-5]*.tar"))  # from *00.tar to *59.tar
     .settings(cache=True)
     .gen(laion=process_webdataset(spec=WDSLaion), params="file")
-    .save()  # materialize chain to avoid downloading data multiple times
 )
 
 meta_pq = (
     DataChain.from_parquet("gs://datachain-demo/datacomp-small/metadata/0020f*.parquet")
-    .filter(
-        C("uid").in_(values[0] for values in wds.select("laion.json.uid").collect())
-    )
-    .map(stem=lambda file: file.get_file_stem(), params=["source.file"], output=str)
-    .save()
+    .settings(cache=True)
+    .mutate(stem=path.file_stem(C("source.file.name")))
 )
 
 meta_emd = (
     DataChain.from_storage("gs://datachain-demo/datacomp-small/metadata/0020f*.npz")
+    .settings(cache=True)
     .gen(emd=process_laion_meta)
-    .filter(
-        C("emd.index").in_(
-            values[0] for values in meta_pq.select("source.index").collect()
-        )
-    )
-    .map(stem=lambda file: file.get_file_stem(), params=["emd.file"], output=str)
+    .mutate(stem=path.file_stem(C("emd.file.name")))
 )
 
-
 meta = meta_emd.merge(
     meta_pq,
     on=["stem", "emd.index"],
     right_on=["stem", "source.index"],
 )
 
-res = wds.merge(meta, on="laion.json.uid", right_on="uid")
+res = wds_images.merge(meta, on="laion.json.uid", right_on="uid").save("wds")
```

Comment:
You would probably save some more time if you omit this save, but not sure if you think it's important to save the final result for this example. When actually doing datacomp, we would probably have no choice but to save here.

Reply:
Let's remove all the save(). Save makes more sense when you actually reuse the results which we don't here.